### PR TITLE
fix:  eliminate misc sphinx warnings...

### PR DESCRIPTION
**Pull request by vwheeler63**
...and miscellaneous doc fixes/oversights from past updates.

- Sphinx issues warnings/errors when section heading underlines are not long enough or uses a character that does not fit the document structure already established.
- The .. code-block:: <lang> updates are for correct syntax coloring. Just plain .. code:: sometimes does not do syntax coloring correctly.
- Also fixed are a few urecognized interpreted-text prefixes, e.g. :refr: is misspelled and must be :ref: or sphinx doesn't know what to do with it.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by kisvegabor — APPROVED** [review]
Oh, god, thank you so much!

File: docs/overview/fs.rst [hunk 1]
```diff
@@ -183,8 +183,8 @@ The implementation is documented below. Note that the FS functions make calls
 to other driver FS functions when the cache is enabled. i.e., ``lv_fs_read`` may call the driver's ``seek``
 so the driver needs to implement more callbacks when the cache is enabled.
 
-``lv_fs_read`` :sub:`(behavior when the cache is enabled)`
--------------------------------------------------
+``lv_fs_read`` :sub:`(Behavior When Cache is Enabled)`
+----------------------------------------------------------
```
Comment (liamHowatt) [inline docs/overview/fs.rst:187]: Is this ok?

**Review by liamHowatt — APPROVED** [review]

File: docs/overview/fs.rst (hunk 1 above)
Comment (vwheeler63) [reply docs/overview/fs.rst:187]: Fair question, and thank you for asking!  The short answer is "no", but I am handling it in a separate PR.  Reason:  on deeper review of the pattern used for capitalization of sub-headings, that most of the .RST files in fact follow the strict Chicago Manual of Style on this, which condones capitalizing (using Title Case) sub-headings only down to sub-heading level 2.  In other words, use *Title Case* on chapter titles, but *Sentence Case* on sub-headings within chapters.  (Some other manuals of style condone capitalizing (using Title Case) down to sub-heading level 5, and that is what was in my head when I did the above.  They are subsequently handled in something I am working on now.

Comment (liamHowatt) [reply docs/overview/fs.rst:187]: The capitalization looks good to me :slightly_smiling_face:  I was wondering about the number of dashes.

Comment (vwheeler63) [reply docs/overview/fs.rst:187]: > The capitalization looks good to me 🙂 I was wondering about the number of dashes.

\*\*laughing at self\*\*  Yes it is okay for it to be longer, but generates a `sphinx` error when it is too short.  This too is fixed (for consistency -- made to equal the length of the title) in subsequent work (not yet submitted).

**Pull request merged**